libs/application-generic/src/services/calculate-delay/compute-job-wait-duration.service.ts

```diff
@@ -37,35 +39,46 @@ export class ComputeJobWaitDurationService {
       stepMetadata.type;
 
     if (digestType === DelayTypeEnum.SCHEDULED) {
-      const delayPath = (stepMetadata as IDelayScheduledMetadata).delayPath;
-      if (!delayPath) throw new ApiException(`Delay path not found`);
-
-      const delayDate = payload[delayPath];
-      const delay = differenceInMilliseconds(new Date(delayDate), new Date());
-
-      if (delay < 0) {
-        throw new ApiException(
-          `Delay date at path ${delayPath} must be a future date`
-        );
+      const userMetadata = this.getUserScheduledDelayMetadata(chimeraResponse);
+      let delay = 0;
+
+      if (userMetadata.date) {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** ‏ will this work if the return of the `this.getUserScheduledDelayMetadata(chimeraResponse);` ends up being `undefined`?

Reply:
could it be a difference between `return;` and `return undefined;`?

Reply:
George from the past created [this one](https://linear.app/novu/issue/NV-3701/application-generic-set-strictnullchecks-as-true)